modules/product/src/main/java/com/opengamma/strata/product/credit/CdsIndex.java

```diff
@@ -0,0 +1,1220 @@
+   * <p>
+   * The fixed notional amount applicable during the lifetime of the CDS.
+   * The currency of the notional is specified by {@code currency}.
+   */
+  @PropertyDefinition(validate = "ArgChecker.notNegativeOrZero")
+  private final double notional;
+  /**
+   * The accrual schedule.
+   * <p>
+   * This is used to define the accrual periods.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final PeriodicSchedule accrualSchedule;
+  /**
+   * The fixed coupon rate.
+   * <p>
+   * This must be represented in decimal form.
+   */
+  @PropertyDefinition(validate = "ArgChecker.notNegative")
+  private final double fixedRate;
+  /**
+   * The day count convention.
+   * <p>
+   * This is used to convert dates to a numerical value.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final DayCount dayCount;
+  /**
+   * The payment on default.
+   * <p>
+   * Whether the accrued premium is paid in the event of a default.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final PaymentOnDefault paymentOnDefault;
+  /**
+   * The protection start of the day.
+   * <p>
+   * When the protection starts on the start date.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final ProtectionStartOfDay protectionStart;
+  /**
+   * The number of days between valuation date and step-in date.
+   * <p>
+   * The step-in date is also called protection effective date. 
+   * It is usually 1 calendar day for standardized CDS index contracts. 
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final DaysAdjustment stepinDateOffset;
+  /**
+   * The number of days between valuation date and settlement date.
+   * <p>
+   * It is usually 3 business days for standardized CDS index contracts.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final DaysAdjustment settlementDateOffset;
+
+  //-------------------------------------------------------------------------
+  /**
+   * Creates an instance.
+   * <p>
+   * The start date adjustment, end date adjustment, and roll convention are switched off. 
+   * Use {@link #builder()} for the full flexibility.
+   * 
+   * @param buySell  buy or sell
+   * @param cdsIndexId  the CDS index ID
+   * @param legalEntityIds  the legal entity IDs
+   * @param currency  the currency
+   * @param notional  the notional 
+   * @param startDate  the start date
+   * @param endDate  the end date
+   * @param paymentFrequency  the coupon frequency
+   * @param businessDayAdjustment  the business day adjustment
+   * @param stubConvention  the stub convention
+   * @param fixedRate  the fixed coupon rate
+   * @param dayCount  the day count convention
+   * @param paymentOnDefault  the payment on default
+   * @param protectStart  the protection start of the day
+   * @param stepinDateOffset  the step-in date offset
+   * @param settlementDateOffset  the settlement date offset
+   * @return the instance
+   */
+  public static CdsIndex of(
```

Comment:
This is too many arguments for an `of` method - users should just use the builder